source/h1_connection.c

```diff
@@ -1641,37 +1696,21 @@ static int s_try_process_next_stream_read_message(struct aws_h1_connection *conn
     size_t bytes_processed = prev_cursor_len - message_cursor.len;
     queued_msg->copy_mark += bytes_processed;
 
+    AWS_ASSERT(connection->thread_data.read_buffer.unprocessed_bytes >= bytes_processed);
```

Comment:
Would we be better off saturate sub-ing this?

Reply:
Hmmmm.
Yeah, I'm always on the fence about using ASSERTS to ensure your math is doing what you intend, vs being safe but being unaware that your math is wrong.

Maybe I'll do both: ASSERT but also saturate
Nothing will explode if these particular numbers are off, but I'd love to fix it if they ever get that way

Reply:
I went and did some more careful considerations of places I was just ASSERTing my math was all correct